tests/mtest/src/main/scala/tests/BaseSuite.scala

```diff
@@ -144,15 +144,26 @@ class BaseSuite extends TestSuite {
 
   def isTestSuiteEnabled: Boolean = true
 
-  def test(name: String)(fun: => Any): Unit = {
+  def test(name: String, expectFailure: Boolean = false)(fun: => Any): Unit = {
```

Comment:
This is an improvement! I am not sure it's a good idea to add such a specific parameter to this signature however. I have wanted to add tag support down the road so that we can classify some tests for example as flaky. What do you think about this approach instead?
```suggestion
class Tag
case object ExpectFailure extends Tag
  def test(name: String, tags: Tag*)(fun: => Any): Unit = {
    val isExpectFailure = tags.contains(ExpectFailure)
```

Reply:
Another design I have thought of could be to replace `name: String` with a custom data structure

```scala
case class TestData(name: String, isExpectedFailure: Boolean, tags: List[Tag]) {
  def isFailing: TestData = copy(isExpectedFailure = true)
  def tag(t: Tags): TestData.copy(tags = t :: tags)
}
object TestData {
  implicit def generate(name: String): TestData = TestData(name, false, Nil)
}
def test(data: TestData)(fun: => Unit): Unit = ...
```

The implicit conversion will make existing tests continue to work unchanged and we can more easily introduce new APIs to tweak the test metadata

```scala
test("issue-42".isFailing.tag(Flaky)) { ... }
```

Reply:
Just validated this compiles and works as expected
<img width="558" alt="Screenshot 2020-01-05 at 17 17 27" src="https://user-images.githubusercontent.com/1408093/71782871-4a5c5900-2fdf-11ea-8fee-3066759d712f.png">

Reply:
I like the design! I've submitted a proposal that implements it